jerry-core/jerry.c

```diff
@@ -2023,42 +2019,19 @@ jerry_parse_and_save_snapshot (const jerry_api_char_t *source_p, /**< script sou
   jsp_status_t parse_status;
   ecma_compiled_code_t *bytecode_data_p;
 
-  snapshot_buffer_write_offset = 0;
-  snapshot_last_compiled_code_offset = 0;
+  snapshot_buffer_write_offset = JERRY_ALIGNUP (sizeof (jerry_snapshot_header_t),
+                                                MEM_ALIGNMENT);
```

Comment:
This simplifies things. There are much more complex things we should worry about.